UnityClient/Assets/Scripts/Components/UI/Minimap/Minimap.cs

```diff
@@ -0,0 +1,296 @@
+using ROIO;
+using System;
+using System.IO;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class Minimap : MonoBehaviour
+{
+    [SerializeField] private Canvas _displayControl;
+    [SerializeField] private RectMask2D _mask;
+    [SerializeField] private RawImage _mapBase;
+    [SerializeField] private Button _btnPlus;
+    [SerializeField] private Button _btnMinus;
+    [SerializeField] private Button _btnMinimize;
+    [SerializeField] private TextMeshProUGUI _coordinates;
+    [SerializeField] private RawImage _playerIndicator;
+
+    [SerializeField] private int m_nZoom = DEFAULT_ZOOM_INDEX;
+    [SerializeField] private float m_fOffSetX = 0.0f;
+    [SerializeField] private float m_fOffSetY = 0.0f;
+    [SerializeField] private int m_nCenterX = 0;
+    [SerializeField] private int m_nCenterY = 0;
```

Comment:
I kinda dislike the hungarian notation and the underscore for private fields, but that's debatable